drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -56,11 +60,24 @@ static inline void msg_done(struct device *dev, unsigned int flags)
 	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
 	I2C_TypeDef *i2c = cfg->i2c;
 
+	if (!LL_I2C_IsEnabledAutoEndMode(i2c))
+	{
+		/* Wait for transfer to complete */
+		while (!LL_I2C_IsActiveFlag_TC(i2c) ){
```

Comment:
Why wait only for transfer to complete when auto end mode is disabled and not always?

Reply:
If auto end mode is activated TC flag will not be set.
Reference manual (RM0316) for F3 says: 
> TC: Transfer Complete (master mode)
This flag is set by hardware when RELOAD=0, AUTOEND=0 and NBYTES data have been
transferred. It is cleared by software when START bit or STOP bit is set.
Note: This bit is cleared by hardware when PE=0.

**So when auto end mode is activated the msg_done function will read like this:**
76: RESTART = 0, so if evaluates to true
77: Wait for stop condition appearing on bus
80: Clear stop flag
done

**With auto end mode disabled:**
63: if evaluates to true
66: wait for transfer complete flag

now if RESTART = 0 (this message should be followed by a stop condition)
70: if evaluates to true
71: stop condition is generated manually
76: if evaluates to true
77: Wait for stop condition appearing on bus

but if RESTART = 1 (don't generate a stop condition)
70: if evaluates to false
76: if evaluates to false
done

Reply:
> If auto end mode is activated TC flag will not be set.

You are right. I missed that.